packages/webapp/src/containers/Finances/saga.js

```diff
@@ -148,6 +151,69 @@ export function* getFarmExpenseTypeSaga() {
   }
 }
 
+export function* addCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_name } = action.custom_expense_type;
+
+  try {
+    const result = yield call(axios.post, expenseTypeUrl, { farm_id, expense_name }, header);
+    if (result) {
+      yield put(enqueueSuccessSnackbar(i18n.t('message:EXPENSE_TYPE.SUCCESS.ADD')));
+      yield call(getFarmExpenseTypeSaga);
+      history.push('/manage_custom_expenses');
+    }
+  } catch (e) {
+    console.log('failed to add new expense type to the database');
+    yield put(enqueueErrorSnackbar(i18n.t('message:EXPENSE_TYPE.ERROR.ADD')));
+  }
+}
+
+export function* updateCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_type_id } = action;
+  const { expense_type_name } = action.custom_expense_type;
+
+  try {
+    const result = yield call(
+      axios.put,
```

Comment:
Odd; I could have sworn this worked before (maybe this PR or the other changed since I first tested?) but with most recent of version of the API PR, this needs to be `patch`; there is no `put` route.

Reply:
Yup good call thanks!